### PR TITLE
Feature/add devise lockable module

### PR DESCRIPTION
**Pull request by adamesj**
This PR introduces Devise's lockable module for the User table. I've configured the devise initializer file to allow for up to 10 failed attempts before the user is redirected to the new_user_unlock page where they can enter their email for unlocking instructions. I also created a sessions controller that inherits from devise's sessions controller in order to handle the logic of redirecting the user when they fail to authenticate with valid credentials (couldn't figure out a cleaner solution, but I am open to any suggestions). A feature spec is included as well.

### CODE REVIEW

File: spec/features/lock_user_account_spec.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
+require 'rails_helper'
+
+RSpec.describe 'User account gets locked', type: :feature do
```
Comment (CZagrobelny) [inline spec/features/lock_user_account_spec.rb]: Woohoo!  Thank you for this awesome feature spec coverage! 💃 🙏 👏 😁

Comment (adamesj) [reply spec/features/lock_user_account_spec.rb]: Ah! I was losing my mind trying to figure out why the messages weren't showing up which lead to the sessions controller solution. Thanks so much for picking that up. 👍  I'll push up the changes soon!

**Comment by adamesj** [comment]
Changes are pushed into the PR! One thing to note is that the user isn't redirected to the new unlock page when the account is locked, but I figured since they should see a flash message along with the 'Didn't receive unlock instructions?' link in the sign in page that should be enough? Thanks again for catching the issue with Devise! 😅

**Comment by CZagrobelny** [comment]
@adamesj this looks fantastic!  Merging! 💃

**Pull request merged**